src/PdfReader/index.tsx

```diff
@@ -472,36 +503,71 @@ export default function usePdfReader(args: ReaderArguments): ReaderReturn {
     });
   }, [state.scale]);
 
+  /**
+   * TODO: Update to work with sub-chapter links
+   */
   const goToPage = React.useCallback(
     async (href) => {
-      const getIndexFromHref = (href: string): number => {
-        const index = manifest?.readingOrder?.findIndex((link) => {
-          return link.href === href;
-        }) as number;
-        if (index < 0) {
-          throw new Error('Cannot find resource in readingOrder');
+      if (isSinglePDF) {
+        const getIndexFromHref = (href: string): number => {
+          const index = manifest?.toc?.findIndex((link) => {
+            return link.href === href;
+          }) as number;
+          if (index < 0) {
+            throw new Error('Cannot find resource in toc');
+          }
+          return index;
+        };
+
+        const resourceIndex = getIndexFromHref(href);
+        if (manifest?.toc && manifest?.toc[resourceIndex]) {
+          const hashPage = getHashPage(manifest?.toc[resourceIndex].href);
+          dispatch({
+            type: 'NAVIGATE_PAGE',
+            pageNum: hashPage,
+          });
+
+          if (state.settings?.isScrolling && hashPage) {
+            document
+              .querySelector(`[data-page-number="${hashPage}"]`)
+              ?.scrollIntoView();
+          }
         }
-        return index;
-      };
-
-      const resourceIndex = getIndexFromHref(href);
-      dispatch({
-        type: 'SET_CURRENT_RESOURCE',
-        index: resourceIndex,
-        shouldNavigateToEnd: false,
-      });
-
-      if (manifest?.readingOrder && manifest?.readingOrder[resourceIndex]) {
-        const startPage = getStartPage(
-          manifest?.readingOrder[resourceIndex].href
-        );
+      } else {
```

Comment:
My understanding of this change is that if the resource is a single resource pdf, we do not call `SET_CURRENT_RESOURCE` and we call `scrollIntoView` to scroll the user to the given page number. Given that understanding, I have two comments:

1. This code is almost the same as the code for a multi-resource pdf. It would be preferrable to have one general `goToHref` function that handles both cases simultaneously. This is how I imagine that functioning:
  - Get the index for the requested resource.
  - If the requested index is different from the current index, dispatch a `SET_CURRENT_RESOURCE` action (we can actually dispatch this no matter what, if the number doesn't change then nothing will happen).
  - Check if the requested resource has a `#page=<number>` hash parameter. If it does, dispatch a `NAVIGATE_PAGE` action to navigate us to that page.
  - If not, check instead fo the `startPage` hash. Navigate similarly to that page if it's present. 
2. These callbacks (`goToPage` in this case) should not have any side effects (like `scrollIntoView`). They should take input, dispatch actions, and optionally return something but not directly change state or manipulate the document in any way. The flow should be `goToPage("somewhere#page=5")` gets called. This in turn dispatches `NAVIGATE_PAGE` with `pageNum: 5`, which updates the local state. Then, a `useEffect` hook can detect this change and scroll the user to the right page. I admit this is a seemingly roundabout way to do something that is already being done (and working) here more directly. However, this separation of actions, state and effects becomes very important when things get more complex (which they inevitably do). If we navigate the user directly here, then in any function we write to navigate the user we have to remember to scroll them to the right location. Imagine we want to add a feature to take to user to their last known reading location. We make a function to get the last known location from storage and then navigate the user. We have to remember to call `scrolIntoView` in a similar manner there. 
Instead, my idea with this architecture is to do this declaratively. When we want to go somewhere, we just dispatch a `NAVIGATE_PAGE` action and can trust that the reducer will update the state and the effect will be called apropriately. 

I see that `NAVIGATE_PAGE` is already implemented, but I don't see any effect actually scrolling the user to the given page. I'm not sure how it is meant to work exactly but I will take a closer look and get back to you. We may need to add an effect to detect changes in the `pageNumber` state and call `scrollIntoView`.